buildSrc/src/main/groovy/nva-publication-api.java-common-conventions.gradle

```diff
@@ -0,0 +1,156 @@
+plugins {
+    // Apply the java Plugin to add support for Java.
+    id 'java'
+    id 'java-library'
+    id 'jacoco'
+    id 'checkstyle'
+    id 'pmd'
+    id 'base'
+    //TODO: use when this is fixed https://github.com/nebula-plugins/gradle-lint-plugin/pull/335
+    //id 'nebula.lint'
+    id 'maven-publish'
+    id "io.freefair.lombok"
+}
+
+group = 'com.github.bibsysdev'
+
+repositories {
+    mavenCentral()
+    maven { url "https://jitpack.io" }
+    maven { url "https://s3.eu-central-1.amazonaws.com/dynamodb-local-frankfurt/release" }
+}
+
+def hamcrestVersion = '2.2'
+def junit5Version = '5.6.0'
+def mockitoVersion = '3.9.0'
+
+dependencies {
+    implementation group: 'org.slf4j', name: 'slf4j-api', version: '1.8.0-beta4'
+
+    testImplementation group: 'org.mockito', name: 'mockito-core', version: mockitoVersion
+    testImplementation group: 'org.hamcrest', name: 'hamcrest', version: hamcrestVersion
+    testImplementation group: 'org.hamcrest', name: 'hamcrest-core', version: hamcrestVersion
+    testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: junit5Version
+    testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-params', version: junit5Version
+    testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: junit5Version
+
+}
+
+test {
+    useJUnitPlatform {
+        excludeTags("RemoteTest")
+    }
+    failFast = true
+    testLogging {
+        events 'skipped', 'passed', 'failed'
+    }
+
+    finalizedBy jacocoTestReport
+    //TODO: investigate the usage of a BuildConfig file with BuildConfig plugin
+    environment "API_HOST", "localhost"
+    environment "API_SCHEME", "https"
+}
```

Comment:
I am not sure if these settings should be defined in the plugin. If yes, then we are taking about a very specialized plugin that cannot be published.